src/sentry/static/sentry/app/views/settings/components/forms/jsonForm.tsx

```diff
@@ -128,12 +129,59 @@ class JsonForm extends React.Component<Props, State> {
     }
   }
 
+  displayFormPanel = (fields: FieldObject[]) => {
+    const fieldsWithVisibleProp = fields.filter(
+      field => typeof field !== 'function' && defined(field?.visible)
+    ) as Array<Omit<Field, 'visible'> & Required<Pick<Field, 'visible'>>>;
+
+    if (fields.length === fieldsWithVisibleProp.length) {
+      const {additionalFieldProps, ...props} = this.props;
+
+      const areAllFieldsHidden = fieldsWithVisibleProp.every(field => {
+        if (typeof field.visible === 'function') {
+          return !field.visible({...props, ...additionalFieldProps});
+        }
+        return !field.visible;
+      });
+
+      return !areAllFieldsHidden;
+    }
+
+    return true;
+  };
+
+  renderFields = ({
+    fields,
+    formPanelProps,
+    title,
+  }: {
+    fields: FieldObject[];
+    formPanelProps: Pick<
+      Props,
+      | 'access'
+      | 'disabled'
+      | 'features'
+      | 'additionalFieldProps'
+      | 'renderFooter'
+      | 'renderHeader'
+    > &
+      Pick<State, 'highlighted'>;
+    title?: React.ReactNode;
+  }) => {
+    const displayFormPanel = this.displayFormPanel(fields);
```

Comment:
Can we rename this `shouldDisplayFormPanel` -- `displayFormPanel` reads like it will display/render the form panel.

Reply:
sure!